### PR TITLE
Archive De-Archive now in table... Needs nicer Styling

### PR DESCRIPTION
**Pull request by mharasic**

### CODE REVIEW

**Review by henderkes — APPROVED** [review]

File: web/src/components/common/DataTable.jsx [hunk 1]
```diff
@@ -351,42 +351,57 @@ export default class BookmarksTable extends Component {
         }
       })
     }
-    if(this.props.columns.indexOf("archive") !== -1){
+    if(this.props.columns.indexOf("dearchive") !== -1){
       columns.push({
-        Header: 'Archive',
+        Header: 'Unarchive',
```
Comment (henderkes) [inline web/src/components/common/DataTable.jsx]: I think we should call it either Dearchive or Unarchive, but not sometimes the first and other times the other

**Review by henderkes — CHANGES_REQUESTED** [review]

File: web/src/components/common/DataTable.jsx (hunk 1 above)
Comment (henderkes) [reply web/src/components/common/DataTable.jsx]: This is the case at multiple places btw, only marked it here

**Review by henderkes — DISMISSED** [review]

File: web/src/components/common/Backend.jsx [hunk 2]
```diff
@@ -441,20 +441,13 @@ class Backend {
...
     async setUserRoles(mail, firstName, lastName, bio, roles){
-      if (await this.putJson('/api/users', {
+      return this.putJson('/api/users', {
         'email': mail,
         'first_name': firstName,
         'last_name': lastName,
         'bio': bio,
         'roles' : roles
-      })) {
-        if (this.mail == mail) {
```
Comment (Drakulix) [inline web/src/components/common/Backend.jsx]: Why did you remove this check? What if I make myself from an admin to a normal user? Every profile update should always trigger a refresh of the profile, if it is mine.

**Review by Drakulix — DISMISSED** [review]

**Review by Drakulix — APPROVED** [review]
